Service/DataService.php

```diff
@@ -2010,4 +2023,101 @@ public function createAndMapIndex(Environment $environment): void
             'name' => $environment->getAlias()
         ]);
     }
+
+    public function getDataLinks(string $contentTypesCommaList, array $businessIds): array
+    {
+        $items = [];
+        $ouuids = [];
+        foreach ($businessIds as $businessId) {
+            if (isset($this->cacheOuuids[$contentTypesCommaList][$businessId])) {
+                $ouuids[$businessId] = $this->cacheOuuids[$contentTypesCommaList][$businessId];
+            } else {
+                $items[] = $businessId;
+                $ouuids[$businessId] = $businessId;
+            }
+        }
+
+        foreach ($this->contentTypeService->getByNames(explode(',', $contentTypesCommaList)) as $contentType) {
+            if ($contentType->getBusinessIdField() && count($ouuids) > 0) {
+                $result = $this->client->search([
+                    'index' => $contentType->getEnvironment()->getAlias(),
+                    'body' => [
+                        'size' => sizeof($ouuids),
+                        '_source' => $contentType->getBusinessIdField(),
+                        'query' => [
+                            'bool' => [
+                                'must' => [
+                                    [
+                                        'term' => [
+                                            '_contenttype' => $contentType->getName()
+                                        ]
+                                    ],
+                                    [
+                                        'terms' => [
+                                            $contentType->getBusinessIdField() => $items
+                                        ]
+                                    ],
+                                ]
+                            ]
+                        ]
+
+                    ],
+                    'size' => 100,
+                    "scroll" => "1m"
+                ]);
+
+                while (count($result['hits']['hits'] ?? []) > 0) {
```

Comment:
We should really start to create a service/object to handle elasticsearch when new code is created. So we are no longer bound to these kind of constructions to handle the requests.